### ISSUE TITLE
Importing Python modules from a select location

### ISSUE THREAD

**Issue by beepboop123133131** [pyinstaller/pyinstaller#3122]
Thought I'd ask, not a lot on this topic.

Let’s say I had three scripts. Main.py (has all imports), 1.py (random script), 2.py (random script).

```
pyinstaller -F --onedir Main.py (80mb)
pyinstaller -F --onedir 1.py (80mb)
pyinstaller -F --onedir 2.py (80mb)
```
This creates 3 folders I then copy 1.exe and 2.exe to Main folder with all dependencies and this runs fine.

image

Two issues are present:

The issue is the size. One file reduces this to 30mb, one folder keeps it at 80mb

More importantly, the exe’s are unable to leave that folder. I’ve had to resort to using shortcuts in Python.

I am following this, which supposedly is a workaround.

My question is, how can I read the imports from a select location with pyinstaller so I can move the executables?

Logic: Compile Main.py using pyinstaller -F –onefile Main.py (15mb).

Create folder at C:\13\ so exe looks like C:\13\Main.exe

link all exe's at C:\13? with  --runtime-tmpdir PATH

I am struggling to find documentation on getting imports from inside the exe or even "one dir" folder. There is plenty on getting imports from the get go, but not much on getting imports from a select location so I can run the exe anywhere. What's more, it is hard to even find the modules in pyinstaller short of decompiling it adding to this confusion.

Perhaps...

```
pyinstaller -F --runtime-tmpdir C:\13 --onefile 2.py

pyinstaller -F --runtime-tmpdir C:\13 --onedir 3.py
```
Though for me no directory is created. How do I go about getting imports from a select location.

Another possible solution:

```
  if getattr(sys, 'frozen', False): 
      os.path.join(sys._MEIPASS, "C:\aa")
```
lets say we wanted pandas there are a few files to choose from...

**Comment by tallforasmurf** [comment]
I think perhaps you are struggling to re-invent Multi-package bundles, which is a feature that once worked and is [still documented](https://pythonhosted.org/PyInstaller/spec-files.html#multipackage-bundles). If that's what you are after, see #1527.

**Comment by beepboop123133131** [comment]
@tallforasmurf  Uh.. that's a shame it's no longer supported.  How would you recommend tackling this issue?  If I compile them in one file I get 30-50mb files.  I could use one folder, though the size is increased and it does not look appealing.  Is there no way to get all exe's to read from an exe in a set location, folder, or zip file as to not have huge files no matter where the exe's are located?

**Comment by beepboop123133131** [comment]
@tallforasmurf  Hmm.. at the very least, is there a way for it to read files from a set directory rather than having exe's having to be present in one dir folder?

**Comment by htgoebel** [comment]
> How would you recommend tackling this issue?

I can't tell you what tallforasmurf is recommending, but my recommendations can be found in #1527. :-)

**Comment by tallforasmurf** [comment]
My recommendation was to examine the MERGE feature in the doc, as it sounded like what @beepboop123133131 wants, or at least produces a similar result. If @beepboop123133131 agrees that it is functionally the same, this issue can be closed as a dup of #1527.

Since MERGE is currently broken (it is not "not supported", it is *supposed* to work) then the next step is to read the extensive discussion in #1527 and maybe have a go at fixing it. Or if it would be valuable to a commercial business to have it work, to offer funding to pay someone to fix it (it shouldn't be more than a 2-day contract for a skilled Pythonista).
